homeassistant/components/upnp/sensor.py

```diff
@@ -175,6 +189,11 @@ def _has_overflowed(self, current_value) -> bool:
         """Check if value has overflowed."""
         return current_value < self._last_value
 
+    @property
+    def device_class(self):
+        """Return the state class for the sensor."""
+        return self.entity_description.state_class
+    
```

Comment:
As above, the property should not be needed. It should be automatically picked up from the `entity_description` by the base class.

```suggestion
```